### DISCUSSION TITLE
Is hydration required for SSR?

### DISCUSSION THREAD

**Discussion by SteveCookTU** [leptos-rs/leptos#532] [Q&A] [answered]
This may be a dumb question but if it isn't required, I can't get it to work.

With only SSR, the HTML is provided as expected but there is no JS or WASM provided. They are not generated in the site root and therefore not provided to the browser.

I guess my confusion may come from how hydration works. I see `mount_to_body` is called within the hydrate function and it includes the main app component in most examples. What is actually built into the WASM lib if the entire application is mounted to body within it which the server should be providing?

**Comment by gbj** [comment, marked as answer]
Not a dumb question at all. The way server-side rendering + hydration works is this: 
1) On the server, code compiled to your native target with feature `ssr` runs your app and "dehydrates" it by rendering it to HTML.
2) In the browser, code compiled to WebAssembly with feature `hydrate` calls `mount_to_body` and "rehydrates" it by adding event listeners and interactivity. (This is where the term "hydrate" comes from.) This basically walks over the tree of your application's view, picking up the nodes that were already rendered to HTML on the server and adding interactivity.

**Reply by SteveCookTU** [reply]
Awesome! Thank you so much, now I see how it splits the two up.
